web/src/lib/components/user-settings-page/partner-settings.svelte

```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
+	import { UserResponseDto, api } from '@api';
+	import CircleAvatar from '../shared-components/circle-avatar.svelte';
+	import Close from 'svelte-material-icons/Close.svelte';
+	import Button from '../elements/buttons/button.svelte';
+	import PartnerSelectionModal from './partner-selection-modal.svelte';
+	import { handleError } from '../../utils/handle-error';
+	import { browser } from '$app/environment';
+	import { onMount } from 'svelte';
+
+	let partners: UserResponseDto[] = [];
+	let showAddUserModal = false;
+
+	$: {
+		if (browser) {
+			if (showAddUserModal) {
+				document.body.style.overflow = 'hidden';
+			} else {
+				document.body.style.overflow = 'auto';
+			}
+		}
+	}
+
+	const handleAddPartner = async () => {
+		try {
+			showAddUserModal = true;
+		} catch (error) {
+			handleError(error, 'Unable to add partner');
+		}
+	};
+
+	const handleRemovePartner = async (user: UserResponseDto) => {
+		try {
+			await api.partnerApi.removePartner(user.id);
+			partners = await loadPartners();
+		} catch (error) {
+			handleError(error, 'Unable to remove partner');
+		}
+	};
+
+	const addUserHandler = async (event: CustomEvent) => {
+		const { selectedUsers }: { selectedUsers: UserResponseDto[] } = event.detail;
+
+		try {
+			for (const user of selectedUsers) {
+				await api.partnerApi.addPartner(user.id);
+			}
+
+			partners = await loadPartners();
+			showAddUserModal = false;
+		} catch (error) {
+			handleError(error, 'Unable to add partners');
+		}
+	};
+
+	const loadPartners = async () => {
+		const { data: sharedWith } = await api.partnerApi.getAllPartners();
+
+		const partners: UserResponseDto[] = [];
+		for (const partner of sharedWith) {
+			const { data: user } = await api.userApi.getUserById(partner.sharedWith);
+			if (user) {
+				partners.push(user);
+			}
+		}
```

Comment:
Can't the user data be included in the response of the `getAllPartners()` request? I've noticed this looped data fetching in a few other places as well, it would be preferred if the server could send this in one request. But not a big issue, since the number of partners is likely to be small.

Reply:
We actually load from the database the `UserEntity` and then just map it to the id. We could definitely just update the response to send the `UserResponseDto` as a list for sharedWith and sharedBy instead of the id + client side querying.

Reply:
Thanks for pointing this out. Now code looks much better